spec/index.html

```diff
@@ -0,0 +1,553 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8" />
+  <title>MiniApp Widget Specification</title>
+
```

Comment:
The title seems not consistent with the contents. If we go with "Specification", I think use case section should be removed?

Also, if it is specification, I think for at least lifecycle part, WebIDL format should be defined.